clowder-framework/clowder
# Extractor Catalog: Extractor Maturity features

## Description

### Problem
We need a way to mark an extractor as "Development" / "Staging" / "Production" to signal to users whether the extractor is ready for public consumption

### Approach
* Add `maturity` field to ExtractorInfo model
* ~~Add an API method to change the maturity: `PUT /extractors/{name}/maturity/{maturity}`~~
* Display the `maturity` field value in the Maturity column on the Extractor Catalog
* Add helper functions to `extractors.js` for fetching / updating ExtractorInfo via API
* Wire up dropdown buttons for "Move to Development" / "Move to Production" / etc to the API

### How to Test
Prerequisites: at least one extractor registered, RabbitMQ plugin enabled, at least one admin user

1. Checkout and run this branch locally
2. Login to Clowder as an admin user
3. From the top navbar, choose Explore > Extractors
    * You should be brought to the Extractor Catalog view
    * You should see that each extractor has a Maturity value in the table
    * Maturity for all extractors should default to "Development"
4. Click the three vertical dots to the right of the extractor in the table
    * You should see the More Actions dropdown menu expand
    * You should see options to "Move to Staging" and "Move to Production"
5. Click "Move to Staging" to update the maturity
    * You should see the view refresh
    * You should see the Maturity value in the table has now updated
6. Expand the dropdown again
    * You should see that "Move to Development" now appears
    * You should see that "Move to Staging" no longer appears in the list

## Review Time Estimate


## Types of changes


## Checklist:
